File: my_class_pkg/src/ros_action_client.cpp
#include <ros/ros.h>
#include <actionlib/client/simple_action_client.h>
#include <actionlib/client/terminal_state.h>
#include <my_class_pkg/MyActionAction.h>
int main(int argc, char **argv) {
  ros::init(argc, argv, "cre_client");
  ros::NodeHandle nh;
actionlib::SimpleActionClient<my_class_pkg::MyActionAction> client("my_action", true);
ROS_INFO("Waiting for action server to start...");
client.waitForServer();
my_class_pkg::MyActionGoal goal;
goal.object_name = "world";

ROS_INFO("Sending goal...");
client.sendGoal(goal);
bool finished_before_timeout = client.waitForResult(ros::Duration(30.0));
if (finished_before_timeout) {
  actionlib::SimpleClientGoalState state = client.getState();
  ROS_INFO("Action finished: %s", state.toString().c_str());
} else {
  ROS_INFO("Action did not finish before the time out.");
}
return 0;
}

File: ykr_ros_class_ws/src/my_class_pkg/scripts/ros_action_client.py
#!/usr/bin/env python3
import rospy
import actionlib

from my_class_pkg.msg import MyActionAction, MyActionGoal, MyActionResult, MyActionFeedback

def feedback_cb(feedback):
    rospy.loginfo('Progress: {}'.format(feedback.progress))

if __name__ == '__main__':
    rospy.init_node('my_client')
    client = actionlib.SimpleActionClient('my_action', MyActionAction)
    
    # 等待服务端启动
    rospy.loginfo('Waiting for action server to start...')
    client.wait_for_server()
    
    # 创建并发送目标
    goal = MyActionGoal()
    goal.object_name = 'world'
    rospy.loginfo('Sending goal...')
    client.send_goal(goal, feedback_cb=feedback_cb)
    
    # 等待结果
    client.wait_for_result()
    
    # 获取并输出结果
    result = client.get_result()
    if result.success:
        rospy.loginfo('Action succeeded')
    else:
        rospy.loginfo('Action failed')
